packages/components/src/color-picker/index.js

```diff
@@ -106,140 +113,137 @@ const dataToColors = ( oldColors, { source, valueKey, value } ) => {
 		source,
 		...{ ...oldColors[ source ], ...{ [ valueKey ]: value } },
 	};
-};
-
-export default class ColorPicker extends Component {
-	constructor( { color = '0071a1' } ) {
-		super( ...arguments );
-		const colors = colorToState( color );
-		this.state = {
-			...colors,
-			draftHex: toLowerCase( colors.hex ),
-			draftRgb: colors.rgb,
-			draftHsl: colors.hsl,
-		};
-		this.commitValues = this.commitValues.bind( this );
-		this.setDraftValues = this.setDraftValues.bind( this );
-		this.resetDraftValues = this.resetDraftValues.bind( this );
-		this.handleInputChange = this.handleInputChange.bind( this );
```

Comment:
These are stable across rerenders. To maintain that in the refactor I think the easiest option is to put them all in a lazily initialized ref. As an example of that, outside the component function something like this:
```javascript
function createColorOperations( setColors ) {
	commitValues: ( data ) => { /* … */ },
	resetDraftValues: () => { /* … */ },
	setDraftValues: ( data ) => { /* … */ },
	handleInputChange: ( data ) => { /* … */ }
	throttledCommitValues = useThrottle( commitValues, 50 );
	return { 
		commitValues,
		resetDraftValues,
		setDraftValues,
		handleInputChange,
		throttledCommitValues
	};
}
```
Then use it inside the component function like so:
```javascript
	const colorOperations = useRef();
	if ( ! colorOperations.current ) {
		colorOperations.current = createColorOperations( setColors );
	}
	const {
		commitValues,
		resetDraftValues,
		setDraftValues,
		handleInputChange,
		throttledCommitValues
	} = colorOperations.current;
```
That will make `commitValues` stable so it can then be throttled as done by the `Saturation` component. On a more general note, `Hue` and `Alpha` should probably also be throttling their use of it and that's why I included creating the throttled version of the function above (a bit like @daddou-ma had suggested https://github.com/WordPress/gutenberg/pull/26770/files#r521300047).

From there I think we could remove the debounce of `onChangeComplete` which was apparently only ever an intent. As Ella suspected https://github.com/WordPress/gutenberg/pull/26770/files#r519931211, it does not work (I've tested from trunk to verify). Beyond that, debounce doesn't seem right for this and throttle seems more appropriate.